eng/scripts/Invoke-Codegeneration.ps1

```diff
@@ -0,0 +1,33 @@
+<#
+.SYNOPSIS
+Invokes Autoest in the specified directory.
+
+.DESCRIPTION
+Invokes Autorest in the specified directory.
+
+This script is all encompassing for running Autorest where it'll first ensure Autorest is installed then run Autorest.
+
+.PARAMETER Directory
+The directory where Autorest will be invoked.
+
+.PARAMETER AutorestOptions
+Additional options to pass to the Autorest command. By default, this will just run 'autorest'.
+#>
+
+param(
+  [Parameter(Mandatory = $true)]
+  [string]$Directory,
+
+  [Parameter(Mandatory = $false)]
+  [string]$AutorestOptions
+)
+
+# Ensure Autorest is installed.
+npm install -s -g autorest
```

Comment:
Should we check $lastexitcode on the npm and autorest call?

Reply:
That's a good idea, I previously was having CI install this a step early but I want to make it so this script can be ran from a clean pull of the repo on a development machine too.

I'll add an exit code check so a better error message can be thrown/logged